addons/binding/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/handler/SinopeThermostatHandler.java

```diff
@@ -0,0 +1,279 @@
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.sinope.SinopeBindingConstants;
+import org.openhab.binding.sinope.internal.config.SinopeConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.tulip.sinope.core.SinopeDataReadRequest;
+import ca.tulip.sinope.core.SinopeDataWriteRequest;
+import ca.tulip.sinope.core.appdata.SinopeHeatLevelData;
+import ca.tulip.sinope.core.appdata.SinopeOutTempData;
+import ca.tulip.sinope.core.appdata.SinopeRoomTempData;
+import ca.tulip.sinope.core.appdata.SinopeSetPointModeData;
+import ca.tulip.sinope.core.appdata.SinopeSetPointTempData;
+import ca.tulip.sinope.core.internal.SinopeDataAnswer;
+import ca.tulip.sinope.util.ByteUtil;
+
+/**
+ * The {@link SinopeThermostatHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Pascal Larin - Initial contribution
+ */
+public class SinopeThermostatHandler extends BaseThingHandler implements ThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(SinopeThermostatHandler.class);
+
+    private SinopeGatewayHandler gatewayHandler;
+
+    public SinopeThermostatHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Channel channel = getThing().getChannel(channelUID.getId());
+        SinopeGatewayHandler gateway = getSinopeGatewayHandler();
+        gateway.stopPoll();
+        try {
+            if (channel != null && SinopeBindingConstants.CHANNEL_SETTEMP.equals(channelUID.getId())
+                    && command instanceof DecimalType) {
+                setSetpointTemp(this, ((DecimalType) command).floatValue());
+            }
+
+            if (channel != null && SinopeBindingConstants.CHANNEL_SETMODE.equals(channelUID.getId())
+                    && command instanceof StringType) {
+                setSetpointMode(this, Integer.parseInt(((StringType) command).toString()));
+            }
+        } catch (IOException e) {
+            logger.error("Cannot handle command for channel {} because of {}", channelUID.getId(),
+                    e.getLocalizedMessage());
+            gateway.setCommunicationError(true);
+
+        }
+        gateway.schedulePoll();
+    }
+
+    public void setSetpointTemp(SinopeThermostatHandler sinopeThermostatHandler, float temp)
+            throws UnknownHostException, IOException {
+        int newTemp = (int) (temp * 100.0);
+        byte[] deviceId = SinopeConfig.convert(sinopeThermostatHandler.getDeviceId());
+        if (deviceId == null) {
+            logger.error("Invalid Device id: {}", deviceId);
```

Comment:
Error messages end up in the users log file so he can do something about it or report a but, so please either describe the error in more detail or set the log message to debug since you set the status below anyway.